adapter/internal/api/apis_impl.go

```diff
@@ -205,9 +220,9 @@ func verifyMandatoryFields(apiJSON config.APIJsonData) error {
 	return nil
 }
 
-// ApplyAPIProject accepts an apictl project (as a byte array) and updates the xds servers based upon the
-// content.
-func ApplyAPIProject(payload []byte, environments []string) error {
+// ApplyAPIProjectFromAPIM accepts an apictl project (as a byte array), list of vhosts with respective environments
+// and updates the xds servers based upon the content.
+func ApplyAPIProjectFromAPIM(payload *[]byte, vhostToEnvsMap map[string][]string) error {
```

Comment:
we could avoid using * along with array?

Reply:
I thought of adding * since if we use []byte, it is passing large []byte, which will copy []byte array from func to func, if we use * then it passes the memory address of same []byte,
@VirajSalaka if it is not a big performance issue, then shall we remove *, WDYT?

Reply:
fixed with https://github.com/wso2/product-microgateway/pull/1826/commits/5811f21ef1af299c2647245d87a2afb5a597f6dc